flytekit/core/promise.py

```diff
@@ -687,15 +687,15 @@ def create_and_link_node(
         )
     )
 
+    node_name = f"{entity.__module__}.{entity.name}"
```

Comment:
do we actually need the module in the node name?

Reply:
Amended the logic, I'm not including the module name here anymore - we'll just handle duplicate entity names naively by incrementing a monotonically increasing count